nncf/quantization/algorithms/weight_compression/weight_lowering.py

```diff
@@ -222,25 +222,22 @@ def calculate_integer_quantization_params(
     assert mode != CompressWeightsMode.NF4, "The function supports integer quantization only"
     num_bits = config.num_bits
 
-    level_low = 0
-    level_high = 2**num_bits - 1
-
     if weight.dtype != TensorDataType.float32:
         weight = weight.astype(TensorDataType.float32)
 
     if mode in [CompressWeightsMode.INT8_ASYM, CompressWeightsMode.INT4_ASYM]:
+        level_low = 0
+        level_high = 2**num_bits - 1
         min_values = fns.min(weight, axis=reduction_axes, keepdims=True)  # [a1, r, a2] -> [a1, 1, a2]
         max_values = fns.max(weight, axis=reduction_axes, keepdims=True)  # [a1, r, a2] -> [a1, 1, a2]
         scale, zero_point = calculate_scale_zero_point(
             min_values, max_values, level_low, level_high, narrow_range=False
         )
     else:
-        level_low_sym = -(2 ** (num_bits - 1))
-        level_high_sym = 2 ** (num_bits - 1) - 1
-
+        level_high = 2 ** (num_bits - 1) - 1
         scale = fns.max(fns.abs(weight), axis=reduction_axes, keepdims=True)  # [a1, r//gs, 1, a2]
-        scale = scale / level_high_sym
-        zero_point = fns.as_tensor_like(scale, [-level_low_sym]).astype(TensorDataType.int32)
+        scale = scale / level_high
+        zero_point = fns.as_tensor_like(scale, [0]).astype(TensorDataType.int32)
```

Comment:
I would suggest to refactor weight_lowering to avoid creation unnecessary tensors and then checking data of tensor in the weight compression transformation.